### ISSUE TITLE
--insert-pragma causing weird commenting issue due to single asterisk comment

### ISSUE THREAD

**Issue by SYGINT** [prettier/prettier#3269]
**Prettier 1.8.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6AVAAgMoAsCWADoflAOYBqAhgDb4AmVM+0SmAOlJlgKJRhVCAZwCuNJnCGYYuOJgButBkxZdo02ZkJUAngFsEMTBELNoU7WTmlMYWWADWEETE7dMACSpSATnABmcH78ktIQmHBUdsb+GnL+ED56UuoycpZyAO74MphUmLI0hEGYBjIQ9AA0mH5CcDDUdIxmUNUw4bX1eTQ0cZgJSQpKzapu6KgglSAmLULIoN4wyP60dVMARj5RDvXY2mCkZMgwPiJwU-QQYMur5yCkdT4wAApbZHpUx6d3AFZCAB4AIS2jl2VAMABlSHAbjQ1tMXIQXAAmWHw7Q+R7IEDrKjrHQ0aCTECEHykGAAdQYMmQAA4AAxTUkQOoUraEL5nKZ+ACOInwfleVHenyQKzhdzqenwnMlhxocAAiiIIPA0XcYHiqfQaUhkVMTlR8HRyABhCB6D7YqDQGFTER1AAqeLmYtuAF93UA)
```sh
--insert-pragma
```

**Input:**
```jsx
/* ShippingValidation: 
 * Encapsulates the validation on the payment options page in checkout
 */
```

**Output:**
```jsx
/**
 * /* ShippingValidation:
 * Encapsulates the validation on the payment options page in checkout
 *
 * @format
 */

```

**Expected behavior:**
```jsx
/** @format */

/* ShippingValidation:
 * Encapsulates the validation on the payment options page in checkout
 */
```

or

```jsx
/**
 * ShippingValidation:
 * Encapsulates the validation on the payment options page in checkout
 *
 * @format
 */
```

or

```jsx
/*
 * ShippingValidation:
 * Encapsulates the validation on the payment options page in checkout
 *
 * @format
 */
```

I'd be fine if it added the proper Doc Block asterisk or if it just kept it a single, either way is fine.
I can TRY to get around to creating a PR for this, but I'm not sure I have the time today.

**Comment by gnarmedia** [comment]
Looks to be an issue with jest-docblock not realizing it's a malformed doc-block, I'll work on a fix for that.

**Comment by lydell** [comment]
This is definitely a bug. Not sure what the expected output is though.

**Comment by gnarmedia** [comment]
@lydell, it's a bug, but the bug is really in jest-docblock, it's desired behavior could be a few things, I think option 1 is the most sensible with the least caveats.

**Comment by gnarmedia** [comment]
Single line comments do even weirder things...

** Input **
```jsx
/* ShippingValidation: 
 */
```

** Output **
```jsx
/**undefined * /* ShippingValidation:undefined *undefined * @formatundefined */
```


** Input **
```jsx
/* ShippingValidation: */
```

** Output **
```jsx
/**undefined * /* ShippingValidation:undefined *undefined * @formatundefined */
```

**Comment by gnarmedia** [comment]
I think it's all just a minor parsing issue in jest-docblock, I'll look into it later

**Comment by aij** [comment]
It's also problematic with eslint and flow.

For example, eslint no-undef reports errors after prettier --insert-pragma changes

```
/* global foo */
```

to

```
/**
 * /* global ccapLiftFunctions
 *
 * @format
 */
```

The `@flow` comments are handled similarly... They at least remain recognizable to flow itself, but (in addition to being ugly) some editor plugins fail to recognize them.

**Comment by fisker** [comment]
Fixed by #13054
